File: data/llama3_8B_ins_results/readable_format_HPEC/fortran/file255.f90
subroutine print_array(ni, nj, a, r, q)
implicit none

DATA_TYPE, dimension(nj, ni) :: a
DATA_TYPE, dimension(nj, nj) :: r
DATA_TYPE, dimension(nj, ni) :: q
integer :: ni, nj
integer :: i, j
do i = 1, ni
do j = 1, nj
write(0, DATA_PRINTF_MODIFIER) a(j, i)
if (mod((i - 1), 20) == 0) then
write(0, *)
end if
end do
end do
write(0, *)
do i = 1, nj
do j = 1, nj
write(0, DATA_PRINTF_MODIFIER) r(j, i)
if (mod((i - 1), 20) == 0) then
write(0, *)
end if
end do
end do
write(0, *)
do i = 1, ni
do j = 1, nj
write(0, DATA_PRINTF_MODIFIER) q(j, i)
if (mod((i - 1), 20) == 0) then
write(0, *)
end if
end do
end do
write(0, *)
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file255.cpp
#include <stdio.h>
#define DATA_TYPE int
#define DATA_PRINTF_MODIFIER "%d "
#define ni 100
#define nj 100

void print_array(DATA_TYPE a[nj][ni], DATA_TYPE r[nj][nj], DATA_TYPE q[nj][ni]) {
    for (int i = 0; i < ni; i++) {
        for (int j = 0; j < nj; j++) {
            printf(DATA_PRINTF_MODIFIER, a[j][i]);
            if (j % 20 == 0)
                printf("\n");
        }
    }
    printf("\n");

    for (int i = 0; i < nj; i++) {
        for (int j = 0; j < nj; j++) {
            printf(DATA_PRINTF_MODIFIER, r[j][i]);
            if (j % 20 == 0)
                printf("\n");
        }
    }
    printf("\n");

    for (int i = 0; i < ni; i++) {
        for (int j = 0; j < nj; j++) {
            printf(DATA_PRINTF_MODIFIER, q[j][i]);
            if (j % 20 == 0)
                printf("\n");
        }
    }
    printf("\n");
}
